docs/datatypes/matrices.md

```diff
@@ -224,6 +227,27 @@ If you have a matrix where the first dimension means `x` and the second
 means `y`, this will look confusing since `x` is printed as _column_ 
 (vertically) and `y` as _row_ (horizontally).
 
+## Not rectangular arrays
+
+By nesting arrays it is possible to have arrays that are not rectangular, for example.
+```js
+[[1, 2], [3, 4]] // rectangular of size [2, 2]
+[[1, 2], [3]]    // jagged
+[[1, 2], 3]      // heterogeneous
+```
+
+These types of arrays can't be converted to a matrix, but many operations are available for them.
```

Comment:
Say "Jagged and heterogeneous arrays" instead of "These types of arrays" because one of the examples you just showed was rectangular and _can_ be converted to a matrix.